packages/mobile/src/send/SendConfirmation.tsx

```diff
@@ -45,251 +48,180 @@ import { navigate } from 'src/navigator/NavigationService'
+function SendConfirmation(props: Props) {
+  const [modalVisible, setModalVisible] = useState(false)
+  const [encryptionDialogVisible, setEncryptionDialogVisible] = useState(false)
+  const [buttonReset, setButtonReset] = useState(false)
+  const [comment, setComment] = useState('')
 
-const mapDispatchToProps = {
-  sendPaymentOrInvite,
-  fetchDollarBalance,
-  fetchDataEncryptionKey,
-}
+  const dispatch = useDispatch()
+  const { t } = useTranslation(Namespaces.sendFlow7)
 
-const mapStateToProps = (state: RootState, ownProps: OwnProps): StateProps => {
-  const { route } = ownProps
-  const { transactionData, addressJustValidated } = route.params
-  const { e164NumberToAddress } = state.identity
-  const { secureSendPhoneNumberMapping } = state.identity
+  const { transactionData, addressJustValidated, currencyInfo } = props.route.params
+  const e164NumberToAddress = useSelector(e164NumberToAddressSelector)
+  const secureSendPhoneNumberMapping = useSelector(secureSendPhoneNumberMappingSelector)
   const confirmationInput = getConfirmationInput(
     transactionData,
     e164NumberToAddress,
     secureSendPhoneNumberMapping
   )
-  const { recipient } = transactionData
-  const addressValidationType = getAddressValidationType(recipient, secureSendPhoneNumberMapping)
+  const {
+    type,
+    amount,
+    recipient,
+    recipientAddress,
+    firebasePendingRequestUid,
+    reason,
+  } = confirmationInput
+  const addressValidationType = getAddressValidationType(
+    transactionData.recipient,
+    secureSendPhoneNumberMapping
+  )
   // Undefined or null means no addresses ever validated through secure send
-  const validatedRecipientAddress = getSecureSendAddress(recipient, secureSendPhoneNumberMapping)
-  const localCurrencyCode = getLocalCurrencyCode(state)
-  const localCurrencyExchangeRate = getLocalCurrencyExchangeRate(state)
-
-  return {
-    account: currentAccountSelector(state),
-    isSending: state.send.isSending,
-    defaultCountryCode: state.account.defaultCountryCode,
-    dollarBalance: state.stableToken.balance || '0',
-    appConnected: isAppConnected(state),
-    transactionData,
-    confirmationInput,
-    addressValidationType,
-    validatedRecipientAddress,
-    addressJustValidated,
-    localCurrencyCode,
-    localCurrencyExchangeRate,
-    isDekRegistered: isDekRegisteredSelector(state) ?? false,
-    addressToDataEncryptionKey: state.identity.addressToDataEncryptionKey,
+  const validatedRecipientAddress = getSecureSendAddress(
+    transactionData.recipient,
+    secureSendPhoneNumberMapping
+  )
+  const account = useSelector(currentAccountSelector)
+  const isSending = useSelector(isSendingSelector)
+  // tslint:disable-next-line: react-hooks-nesting
+  const dollarBalance = useSelector(stableTokenBalanceSelector) || '0'
+  const appConnected = useSelector(isAppConnected)
+  const isDekRegistered = useSelector(isDekRegisteredSelector) ?? false
```

Comment:
Is the tslint disable not needed here?

Reply:
I guess it's not considered a conditional operator?